### DISCUSSION TITLE
Changing row order with drag&drop

### DISCUSSION THREAD

**Discussion by serhanekicii** [icflorescu/mantine-datatable#89] [Ideas]

**Comment by icflorescu** [comment]
The topic has been discussed before and to be honest I'm not sure if/how this could be implemented without breaking the already existing features (such as filtering, sorting, pagination, etc).

**Reply by otobot1** [reply]
Yeah this seems really tricky. Perhaps it could be supported similarly to filtering, etc where most of the details are left to a parent component. If a "canDrag" prop is set, then the table can call a function passed in from the parent and provide the selected record id, and the ids of the records whose midlines are above and below the current mouse position. The function would then work alongside any other sorting/filtering in the parent component to adjust the records being passed into the table. I suspect it would take a decent bit of testing to ensure everything works correctly and doesn't cause too many renders while moving rows, but I think it could be doable.

**Reply by otobot1** [reply]
I have very little bandwidth these days for my side projects though, so I don't intend to create a PR for this any time soon. If anyone else wants to give it a go, please feel free!
